### PR TITLE
fix: preserve empty accessibility names

### PR DESCRIPTION
**Pull request by vicksiyi**
## Summary
- Preserved ax_node.name whenever it is not None and added regression coverage for empty string vs missing accessibility names.

## Tests
- uv run ruff check browser_use/dom/views.py tests/ci/test_dom_interacted_element.py passed
- uv run ruff format --check browser_use/dom/views.py tests/ci/test_dom_interacted_element.py passed
- git diff --check passed
- python3 -m py_compile browser_use/dom/views.py tests/ci/test_dom_interacted_element.py passed
- uv run pytest tests/ci/test_dom_interacted_element.py blocked by local pytest startup segfault

## Notes
- Related issue: https://github.com/browser-use/browser-use/issues/5041

---
## Summary by cubic
Fixes DOM element mapping and hashing to preserve empty accessibility names, with regression tests. We keep an empty `ax_node.name` and include it in hashes, using `None` only when the name is missing.

- **Bug Fixes**
  - Preserve empty `ax_node.name` in `load_from_enhanced_dom_tree` and include it in `compute_stable_hash`/`__hash__` to distinguish empty vs missing.
  - Add CI tests to verify empty names are preserved and affect hashes.

<sup>Written for commit 863e99c14fe76395196a43da44d1738ae850a5be. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/browser-use/browser-use/pull/5081?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: browser_use/dom/views.py [hunk 1]
```diff
@@ -1022,7 +1022,7 @@ def to_dict(self) -> dict[str, Any]:
 	def load_from_enhanced_dom_tree(cls, enhanced_dom_tree: EnhancedDOMTreeNode) -> 'DOMInteractedElement':
 		# Extract accessibility name if available
 		ax_name = None
-		if enhanced_dom_tree.ax_node and enhanced_dom_tree.ax_node.name:
+		if enhanced_dom_tree.ax_node and enhanced_dom_tree.ax_node.name is not None:
```
[the remark this answers is not in the document]
Comment (vicksiyi) [reply browser_use/dom/views.py:1025]: Addressed in 2b656ee6. `compute_stable_hash` and `__hash__` now both treat an empty accessibility name as present (`name is not None`), and the regression test now verifies empty vs missing accessibility names hash differently.

Validation: `python3 -m py_compile browser_use/dom/views.py tests/ci/test_dom_interacted_element.py`, `uv run ruff check browser_use/dom/views.py tests/ci/test_dom_interacted_element.py`, `uv run ruff format --check browser_use/dom/views.py tests/ci/test_dom_interacted_element.py`, and direct `uv run python` invocation of the focused tests passed. Local `uv run pytest tests/ci/test_dom_interacted_element.py -q` still exits 139 during pytest startup, matching the existing local pytest blocker.

**Comment by vicksiyi** [comment]
Rebased this PR onto the latest `main` in 863e99c14 so it is no longer behind the upstream branch.

Validation: `.venv/bin/ruff check browser_use/dom/views.py tests/ci/test_dom_interacted_element.py`, `.venv/bin/ruff format --check browser_use/dom/views.py tests/ci/test_dom_interacted_element.py`, `python3 -m py_compile browser_use/dom/views.py tests/ci/test_dom_interacted_element.py`, and `git diff --check` pass locally. Focused pytest still exits 139 during local startup, matching the earlier local environment blocker.
